components/ringcentral/sources/new-event/new-event.js

```diff
@@ -0,0 +1,71 @@
+const notificationTypes = require("../common/notification-types");
+const common = require("../common/http-based");
+
+module.exports = {
+  ...common,
+  key: "ringcentral-new-event",
+  name: "New Event (Instant)",
+  description: "Emits an event for each notification from RingCentral of a specified type",
+  props: {
+    ...common.props,
+    extensionId: {
+      optional: true,
+      propDefinition: [
+        common.props.ringcentral,
+        "extensionId",
+      ],
+    },
+    deviceId: {
+      optional: true,
+      propDefinition: [
+        common.props.ringcentral,
+        "deviceId",
+        c => ({ extensionId: c.extensionId }),
+      ],
+    },
+    notificationTypes: {
+      type: "string[]",
+      label: "Notification Types",
+      description: "The types of notifications to emit events for",
+      options({ page = 0 }) {
+        if (page !== 0) {
+          return [];
+        }
+
+        return notificationTypes.map(({ label, key }) => ({
+          label,
+          value: key,
+        }));
+      },
+    },
+  },
+  methods: {
+    ...common.methods,
+    _getEventTypeFromFilter(eventFilter) {
+      return eventFilter
+        .replace(/\/restapi\/v\d+\.\d+\//, "")
+        .replace(/account\/.*?\//, "")
```

Comment:
why the `?`

Reply:
It's just the "non-greedy" regex command, otherwise the `.*` will match every character up until the **last** `/`. Using `.*?` instead of `.*` will match every character up until the **first** `/`.

Reply:
Oh interesting! (I'd just write `/account\/[^/]*\//` which does the same thing right?)

Reply:
(I don't mean to change it, just sharing 😛)